src/Config/CliBuilder.ts

```diff
@@ -89,12 +94,33 @@ export class CliBuilder {
       .reduce((acc: Argv, item: CommandModule) => {
         const handler = item.handler.bind(item);
 
-        item.handler = (args) =>
-          runWithAsyncContext(() => {
-            setContext('args', args);
+        item.handler = async (args: Arguments) => {
```

Comment:
Let's refactor this logic extracting into a class member.

Reply:
Not sure I got the point. Could you please explain?

Reply:
After your changes, the `build` method has become significantly larger and more complex. Please consider refactoring by extracting the `handle` function.